js/client/modules/@arangodb/process-utils.js

```diff
@@ -1554,7 +1555,18 @@ function checkClusterAlive(options, instanceInfo, addArgs) {
   instanceInfo.arangods.forEach(arangod => {
     // agents don't support the ID call...
     if ((arangod.role !== "agent") && (arangod.role !== "single")) {
-      const reply = download(arangod.url + '/_db/_system/_admin/server/id', '', makeAuthorizationHeaders(instanceInfo.authOpts));
+      let reply;
+      try {
+        reply = download(arangod.url + '/_db/_system/_admin/server/id', '', makeAuthorizationHeaders(instanceInfo.authOpts));
+      } catch (e) {
+        print(RED + Date() + " error requesting agent '" + JSON.stringify(arangod) + "' Error: " + JSON.stringify(e) + RESET);
+        if (e instanceof ArangoError && e.message.search('Connection reset by peer') >= 0) {
+          internal.sleep(5);
+          reply = download(arangod.url + '/_db/_system/_admin/server/id', '', makeAuthorizationHeaders(instanceInfo.authOpts));
```

Comment:
So, if we got a "Connection reset by peer" error, we simply try again after 5 seconds?
How is this supposed to work if the peer is already gone?

Reply:
The intention here is, that the system for some reason is not yet fully started up - so we try again.

Reply:
If something isn't fully started up, we should not get a "Connection reset by peer" error message.
We should get "Connection refused" or such, but "reset by peer" indicates that the connection was already established before.